Migrate '/' division operator to use math.div() and list.slash() instead

enbo opened the pull request:
Migrated using https://github.com/sass/migrator.

More info at https://sass-lang.com/documentation/breaking-changes/slash-div.

enbo committed:
Migrate '/' division operator to use math.div() and list.slash() instead

Migrated using https://github.com/sass/migrator.

More info at https://sass-lang.com/documentation/breaking-changes/slash-div.

apexskier commented:
Thanks! It looks like `math.div` is available starting at v1.33, and this project is still at v1.32 (making this a breaking change). You'll need to upgrade the sass npm dependency to address that.

For linting, you can run `yarn fix` to apply prettier.

enbo committed:
Upgrade sass dependency to ^1.34.1

Making math.div() and list.slash() compatible (requires at least Dart Sass 1.33.0).

Also applied Prettier.

enbo commented:
Woops, thank you for catching that. I think fixed the dependency (is it bad practice to set the dependency at a later version than is minimally needed?) and Prettified it.

apexskier commented:
> is it bad practice to set the dependency at a later version than is minimally needed?

Nah, not in my opinion for this. Since it's a dev dependency here it won't actually affect anyone using the package. The actual sass changes will still work back to the oldest supported version.

This package currently doesn't actually do anything to inform the user about sass compatibility. There's not really a way to do this in an automated fashion though. For normal dependencies you can use [`peerDependencies`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#peerdependencies), but in this case that would require `sass` in the production dependency set, which is usually wrong (we'd need `peerDevDependencies`).

It's probably worth adding the minimum supported sass version to the readme, though I wouldn't consider it a blocking thing for this PR. We could even modify and expand the tests to test on old versions of sass, though that might be overkill.

> Minimum supported Sass version
> 
> | Sass Compiler | Version |
> | --- | --- |
> | Dart Sass | 1.33.0 |
> | LibSass | ? |
> | Ruby Sass | ? |

apexskier commented on src/_conversions.scss:
```diff
@@ -1,3 +1,5 @@
+@use "sass:math";
```
I'm surprised this passes prettier. I'll want to check that out.

enbo replied:
Because of the double quotes?

If so, it looks like [Sass's own documentation](https://sass-lang.com/documentation/at-rules/use) is inconsistent / doesn't care about double ("src/corners") vs. single ('foundation/code'), and for Prettier we'd need to use the [Quotes option](https://prettier.io/docs/en/options.html#quotes).

Unrelated, I guess maybe if one wants to be a good citizen alphabetizing the modules would be proper.

apexskier replied:
Looks like it's a bug to me https://github.com/prettier/prettier/issues/11053

apexskier committed:
Merge branch 'master' into master

apexskier committed:
Minor syntax cleanup
